mmaction/models/backbones/resnet_tin.py

```diff
@@ -0,0 +1,370 @@
+import torch
+import torch.nn as nn
+from mmcv.ops import tin_shift
+
+from ..registry import BACKBONES
+from .resnet_tsm import ResNetTSM
+
+
+def linear_sampler(data, offset):
+    """Differentiable Temporal-wise Frame Sampling, which is essentially a
+    linear interpolation process.
+
+    It gets the feature map which has been split into several groups
+    and shift them by different offsets according to their groups.
+    Then compute the weighted sum along with the temporal dimension.
+
+    Args:
+        data (torch.Tensor): Split data for certain group in shape
+            [N, num_segments, C, H, W].
+        offset (torch.Tensor): Data offsets for this group data in shape
+            [N, num_segments].
+    """
+    # [N, num_segments, C, H, W]
+    n, t, c, h, w = data.shape
```

Comment:
use upper case for n, t, c, h, w

we can use lowercase as the iteration variable